command/server.go

```diff
@@ -857,22 +861,31 @@ func (c *ServerCommand) Reload(configPath []string) error {
 
 	// Ensure at least one config was found.
 	if config == nil {
-		retErr := fmt.Errorf("No configuration files found")
-		c.Ui.Error(retErr.Error())
-		return retErr
+		reloadErrors = multierror.Append(reloadErrors, fmt.Errorf("No configuration files found"))
+		goto audit
 	}
 
-	c.reloadFuncsLock.RLock()
-	defer c.reloadFuncsLock.RUnlock()
-
-	var reloadErrors *multierror.Error
 	// Call reload on the listeners. This will call each listener with each
 	// config block, but they verify the address.
 	for _, lnConfig := range config.Listeners {
 		for _, relFunc := range (*c.reloadFuncs)["listener|"+lnConfig.Type] {
 			if err := relFunc(lnConfig.Config); err != nil {
-				retErr := fmt.Errorf("Error encountered reloading configuration: %s", err)
-				reloadErrors = multierror.Append(retErr)
+				reloadErrors = multierror.Append(reloadErrors, fmt.Errorf("Error encountered reloading configuration: %s", err))
```

Comment:
Can we have a `goto audit` after this line despite the label following immediately after this block. This may ensure accidental movement of code to behave properly?